src/pj_tsfn.c

```diff
@@ -2,9 +2,15 @@
 #include <math.h>
 #include <projects.h>
 
-	double
-pj_tsfn(double phi, double sinphi, double e) {
-	sinphi *= e;
-	return (tan (.5 * (M_HALFPI - phi)) /
-	   pow((1. - sinphi) / (1. + sinphi), .5 * e));
+double pj_tsfn(double phi, double sinphi, double e) {
+    sinphi *= e;
+
+    /* sinphi * e == -1 results in zero-division below,
+       instead we return a defined value that indicates
+       something went wrong. */
+    if (sinphi == -1.0)
```

Comment:
I don't trust floating point logic, so I would replace this test by 1. + sinphi == 0.0 to have exactly the same expression as below. Or create a divisor = 1. + sinphi variable and use it in both places